### ISSUE TITLE
Data returned from entry GraphQL mutations is incorrect

### ISSUE THREAD

**Issue by wadamek65** [spicywebau/craft-neo#509]
#### Description
When running a GraphQL mutation on an entry and requesting back the `blockContent` data, this data is incorrect and inconsistent with what the query returns. Any `blockContent` returned will always have empty arrays in `children` and `blockChildren` fields even though the entry query returns the properly.

#### Steps to reproduce
1. Have any content entry with neo blocks.
2. Run a mutation against it and try to create a new block as a child of another block.
3. See that the `blockContent` data is incorrect.

### Screenshots
Below are screenshots of the bug reproduction. First one is what query returns, second one is what mutation returns.

<details>
Query

![image](https://user-images.githubusercontent.com/36732892/139028240-153a605a-45fd-4c08-8100-231388bfb0b1.png)

Mutation

![image](https://user-images.githubusercontent.com/36732892/139028225-0d0d64bd-fe78-4214-ad65-cddb09b852bf.png)
</details>

#### Other information
- Neo version: 2.11.15
- Craft version: 3.7.18
- (if multi-site) What is the affected Neo field's propagation method:
- Is eager-loading used?

**Comment by wadamek65** [comment]
@ttempleton @membla Yes, this occurs only when using the `x-craft-token`. Here is a minimal reproduction that I managed to create.

Mutation
```graphql
mutation SaveSlideBlockContentMutation($id: ID!, $blockContent: blockContent_NeoInput!) {
  save_slides_slide_Entry(id: $id, blockContent: $blockContent) {
    id
      blockContent(level: 0) {
        ...BlockContentBlockFragment
        __typename
    }
    __typename
  }
}


fragment BlockContentBlockFragment on blockContent_NeoField {
  ...on blockContent_image_BlockType {
    id
  }
  ...on blockContent_list_BlockType {
    id
    children {
        id
        __typename
    }
  }
  ...on blockContent_container_BlockType {
    id
    children {
        id
        __typename
    }
  }
  __typename
}
```

Variables:
```json
{
    "id": "625",
    "blockContent": {
        "blocks": [
            {
                "container": {
                    "id": "new-0000",
                    "blockFlexDirection": "COLUMN",
                    "blockAlignItems": "STRETCH",
                    "blockJustifyContent": "FLEX_START",
                    "level": 3
                }
            },
            {
                "image": {
                    "id": "new-0001",
                    "blockObjectFit": "CONTAIN",
                    "level": 4
                }
            }
        ],
        "sortOrder": [
            "685",
            "686",
            "739",
            "740",
            "979",
            "980",
            "new-0000",
            "new-0001"
        ]
    }
}
```

Headers:
```
x-craft-token: fsD7eC5-mK42r5Ep4nhV0sWY2Gp9IZ8X
Authorization: Bearer N3fKGGxINVL2MkiYvBrknHh5e2P0WwZ7
```

The returned data:
```json
{
    "data": {
        "save_slides_slide_Entry": {
            "id": "684",
            "blockContent": [
                {
                    "id": "685",
                    "children": [],
                    "__typename": "blockContent_container_BlockType"
                },
                {
                    "id": "686",
                    "children": [],
                    "__typename": "blockContent_list_BlockType"
                },
                {
                    "id": "739",
                    "children": [],
                    "__typename": "blockContent_container_BlockType"
                },
                {
                    "id": "740",
                    "__typename": "blockContent_image_BlockType"
                },
                {
                    "id": "979",
                    "children": [],
                    "__typename": "blockContent_container_BlockType"
                },
                {
                    "id": "980",
                    "__typename": "blockContent_image_BlockType"
                },
                {
                    "id": "991",
                    "children": [],
                    "__typename": "blockContent_container_BlockType"
                },
                {
                    "id": "992",
                    "__typename": "blockContent_image_BlockType"
                }
            ],
            "__typename": "slides_slide_Entry"
        }
    }
}
```

As you can see, the returned children for lists and containers are empty even though the blocks are correctly created and nested as well. The query does return correct data after running such mutation.

**Comment by ttempleton** [comment]
Thanks @wadamek65 and @membla. I've been able to reproduce it, not sure why it's happening yet but will keep looking at it.

**Comment by ttempleton** [comment]
Just released Neo 2.11.17 which includes a fix for this.
